configure

```diff
@@ -155,6 +156,7 @@ case "$1" in
       echo '    [--without-neon]            Compiles without ARM Neon SIMD instruction set' | tee -a configure.log
       echo '    [--with-dfltcc-deflate]     Use DEFLATE CONVERSION CALL instruction for compression on IBM Z' | tee -a configure.log
       echo '    [--with-dfltcc-inflate]     Use DEFLATE CONVERSION CALL instruction for decompression on IBM Z' | tee -a configure.log
+      echo '    [--with-low-mem]            Build for low memory environments (reduced performance)' | tee -a configure.log
```

Comment:
Would it be clearer if the text was something like this?
"Reduced memory usage for special cases (reduces performance)"

Reply:
Likewise for the other messages of course.

Reply:
@nmoinvaz ^

Reply:
Sorry. I'm not really a fan of this option but I can continue to add it. 

Should I rename the option to `--with-reduced-mem` too?

Reply:
Oh, I didn't know that.

I agree that this option should generally not be used except in certain cases (such as a very very busy nginx proxy for example). It is an useful option, just for a very narrow few niches.

So I think that it should be tagged as an advanced option in cmake.
And perhaps it should not show in the summary unless enabled (Also goes for a few of the other options we'd rather not people use unless they really know what they are doing).

Reply:
And yes, it would be smart to rename the option as well 😄

Reply:
Done.